insights/parsers/satellite_postgresql_query.py

```diff
@@ -0,0 +1,174 @@
+        searching; see its documentation for usage details. If no search
+        parameters are given, no rows are returned.
+
+        It simplify the value of the column according to actual usage.
+
+        Returns::
+
+            list: A list of dictionaries of rows that match the given
+            search criteria.
+
+        Examples::
+
+            >>> query.search(name__startswith='abc') == [
+            ... {'name': 'abc', 'url': '', 'value': 'test'},
+            ... {'name': 'abcdef', 'url': '', 'value': 'test2'}
+            ... ]
+            True
+            >>> query.search(name__startswith='abc', value='test') == [
+            ... {'name': 'abc', 'url': '', 'value': 'test'}
+            ... ]
+            True
+        """
+
+        return keyword_search(self, **kwargs)
+
+
+@parser(Specs.satellite_settings)
+class SatelliteAdminSettings(SatellitePostgreSQLQuery):
+    """
+    Parse the output of the command ``psql -d foreman -c '"select name, value, "default" from settings where name in ('destroy_vm_on_host_delete', 'unregister_delete_host') --csv"``.
+
+    Sample output::
+
+        name,value,default
+        unregister_delete_host,"--- true
+        ...","--- false
+        ..."
+        destroy_vm_on_host_delete,,"--- true
+        ..."
+
+    Examples::
+
+        >>> type(table)
+        <class 'insights.parsers.satellite_postgresql_query.SatelliteAdminSettings'>
+        >>> table.get_setting('unregister_delete_host')
+        True
+        >>> table.get_setting('destroy_vm_on_host_delete')
+        True
+    """
+
+    def _parse_yaml(self, value):
+        if value:
+            try:
+                return yaml.safe_load(value)
+            except Exception:
+                raise ParseException("Bad format value: %s" % value)
+        return value
+
+    def parse_content(self, content):
+        """
+        The "default" and "value" columns must be selected, or else the
+        settings value can't be determined.
+        The "default" and "value" column are in yaml format, it is transfer to
+        python boolean value.
+
+        Raises::
+
+            SkipException: when value or default column isn't found in the
+                            table.
+            ParseException: when the value or default in bad yaml format.
+        """
+        super(SatelliteAdminSettings, self).parse_content(content)
+        if not all(item in self.get_columns() for item in ['default', 'value']):
+            raise SkipException('No default, value columns in the table.')
+        for row in self:
+            row['default'] = self._parse_yaml(row['default'])
+            row['value'] = self._parse_yaml(row['value'])
+
+    def get_setting(self, setting_name):
+        """
+        Get the actual value of setting_name.
+        If the value column isn't empty, the value of the setting_name is the
+        value column, or else it's the default column.
+        """
```

Comment:
Also rather than implicitly returning `None` here, wouldn't it be better to return an empty `list`?

Reply:
> @huali027 please include the `Parameters:` and `Returns:` information in the docstring for users of this methods.

Sure, I'll add it, thanks.

Reply:
> Also rather than implicitly returning `None` here, wouldn't it be better to return an empty `list`?

The return value will be boolean if the setting exists in the table, so I think it's okay to return None if the setting doesn't exist, does it make sense?

Reply:
In the case of the various return types, better add another method `has_setting` to check if the specified setting in the table. And raise a "KeyError" when there is no such setting in the table for the `get_setting`. Just like what the `[IniConfigFile](https://github.com/RedHatInsights/insights-core/blob/master/insights/core/__init__.py#L1382)` do. How do you think?

Reply:
@xiangce, good to me, I'll update it.